MoEPluginBase/MoEPluginController.swift

```diff
@@ -0,0 +1,296 @@
+//
+//  MoEPluginController.swift
+//  MoEPluginBase
+//
+//  Created by Rakshitha on 23/06/22.
+//
+
+import Foundation
+import MoEngageSDK
+import MoEngageInApps
+import UserNotifications
+import CoreText
+
+class MoEPluginController: NSObject {
+    var identifier: String
+    var messageHandler: MoEMessageQueueHandler
+    
+    init(identifier: String){
+        self.identifier = identifier
+        messageHandler = MoEMessageQueueHandler(identifier: identifier)
+    }
+    
+    func pluginInitialized(accountInfo: [String: Any]) {
+        messageHandler.flushAllMessages()
+    }
+    
+    func setBridgeDelegate(delegate: MoEPluginBridgeDelegate) {
+        messageHandler.setBridgeDelegate(delegate: delegate)
+    }
+    
+    func initializeDefaultInstance(sdkConfig: MOSDKConfig, sdkState: Bool, launchOptions:[String: Any]) {
+        #if DEBUG
+        MoEngage.sharedInstance().initializeDefaultTestInstance(with: sdkConfig, andLaunchOptions: launchOptions)
+        #else
+        MoEngage.sharedInstance().initializeDefaultLiveInstance(with: sdkConfig, andLaunchOptions: launchOptions)
+        #endif
+        
+        handleSDKState(sdkState)
+
+        setDelegates()
+    }
+    
+    func initializeInstance(sdkConfig: MOSDKConfig, sdkState: Bool, launchOptions:[String: Any]) {
+        #if DEBUG
+        MoEngage.sharedInstance().initializeTestInstance(with: sdkConfig, andLaunchOptions: launchOptions)
+        #else
+        MoEngage.sharedInstance().initializeLiveInstance(with: sdkConfig, andLaunchOptions: launchOptions)
+        #endif
+        
+        handleSDKState(sdkState)
+
+        setDelegates()
+    }
+    
+    private func setDelegates() {
+        setMoEngageDelegates()
+        setNotificationDelegate()
+    }
+    
+    private func setMoEngageDelegates() {
+        MOMessaging.sharedInstance.setMessagingDelegate(self, forAppID: identifier)
+        MOInApp.sharedInstance().setInAppDelegate(self, forAppID: identifier)
+    }
+    
+    private func setNotificationDelegate() {
+        if UIApplication.shared.isRegisteredForRemoteNotifications {
+            if UNUserNotificationCenter.current().delegate == nil {
```

Comment:
@RakshithaAcharya What's your thought on making it if let?

```
if let delegate = UNUserNotificationCenter.current().delegate {
MoEngage.sharedInstance().registerForRemoteNotification(withCategories: nil, withUserNotificationCenterDelegate: delegate)
} else {
MoEngage.sharedInstance().registerForRemoteNotification(withCategories: nil, withUserNotificationCenterDelegate: self)
}
```

Reply:
@deepa-iOS-SDK addressed